### ISSUE TITLE
C++ compiler "CL.exe" Minor and Macro version aren't detected correctly

### ISSUE THREAD

**Issue by abbas-sabra-sonarsource** [SonarSource/sonarlint-visualstudio#1289]
### Description

`cl.exe` version has a Major, Minor, and Micro values. Currently, only the major part is detected correctly. The problem is that only the platform toolset is used(`ConvertPlatformToolset`) to detect the version while different compiler versions can be used with the same toolset. The compiler version should be detected directly instead of deducing it from the platform.

### Repro steps
Install 19.23.28107 c++ compiler (or any compiler other than the hardcoded ones in `ConvertPlatformToolset`)  on v142 and check that the detected version will always be 19.20.00.

### Expected behavior
The installed version of the compiler is detected. 
The two macros defined later on based on the compiler version should be set to the right value:
```
predefines.Append("#define _MSC_FULL_VER ").Append(probe.Major).Append(probe.Minor).Append(probe.Micro).Append("\n");
predefines.Append("#define _MSC_VER ").Append(probe.Major).Append(probe.Minor).Append("\n");
```

### Actual behavior

The compiler version is always set to a hardcoded value which leads to wrong values of the two mentioned macros.

### Known workarounds

No workaround

#### Questions
Can the compiler version change per project? Yes, but not straightforward. Corner case - maybe ignore?
